### ISSUE TITLE
Construct a Parser object

### ISSUE THREAD

**Issue by hop-along-polly** [xavier643/fluent-csv-php-parser#7]
As a developer I would like yo construct an Object that will abstract the internal workings of parsing a CSV File.

# Requirements
1. The constructor requires a string as it's first parameter representing the path to a file.
2. The constructor optionally accepts a second parameter representing the delimiter for values. If no value is specified the delimiter should default to a comma.
3. The constructor optionally accepts a third argument, an array of strings that defines the header columns.
4. A type of CSVParser\Parser is returned on successful construction of the object.
5. An Exception should be thrown when the file path specified points to a file that does not exist.
6. Valid delimiters should be limited to commas, pipes '|' or tab '\t' characters

# Interface
## Example 1
```php
$parser = new CSVParser\Parser('/tmp/sample.csv');
```
## Example 2
```PHP
$parser = new CSVParser('/tmp/sample.csv', '|');
```

## Example 3
```
$parser = new CSVParser\Parser('/tmp/sample.csv', '|', ['col_1', 'col_2']);
```

**Comment by xavier643** [comment]
This this asking for the constructor to be able to take in 3 different possible options?  Similar to the idea of overloading in other languages?

**Comment by hop-along-polly** [comment]
Correct. PHP does not support method overloading though so you will need to look into default parameters.

**Comment by xavier643** [comment]
This issue has been completed.
